### PR TITLE
2.9 cleanup / update prep

### PR DESCRIPTION
**Pull request by UncleClapton**
* split up and optimize user menu
* adjust user menu transitions
* Clean up some troublesome redux selectors
* Prepare `CHANGELOG` for next version
* Remove teespring link from webstore
* bump version 2.9.0

### CODE REVIEW

File: client/scss/components/_user-menu.scss [hunk 1]
```diff
@@ -74,20 +74,17 @@ $user-menu-border-width-mobile: 5px;
     }
 
     @media not screen and (max-width: $mobile-max-width) {
-      right: calc(#{$avatar-width} / 2 + #{$avatar-margin});
```
Comment (314numberpi) [inline client/scss/components/_user-menu.scss:77]: Why was this change necessary? If the margin on .user-menu changes, menu will no longer adjust.

**Review by 314numberpi — CHANGES_REQUESTED** [review]

File: client/scss/components/_user-menu.scss (hunk 1 above)
Comment (UncleClapton) [reply client/scss/components/_user-menu.scss:77]: 😕 

WHHOOOPPPSSSS. That wasn't meant to stay :/

I set it to exact values so I could mess with the position of the menu, but never moved it back :/

File: client/scss/components/_user-menu.scss [hunk 2]
```diff
@@ -178,8 +175,11 @@ $user-menu-border-width-mobile: 5px;
   @media not screen and (max-width: $mobile-max-width) {
     #UserMenuControl:checked {
       ~ menu {
-        max-width: 100vw;
-        max-height: 100vh;
+        top: #{($avatar-size / 2) + 1rem};
```
Comment (314numberpi) [inline client/scss/components/_user-menu.scss:178]:
```suggestion
        top: #{($avatar-size / 2) + $avatar-top-margin + 0.5rem};
```

File: client/scss/components/_user-menu.scss [hunk 3]
```diff
@@ -178,8 +175,11 @@ $user-menu-border-width-mobile: 5px;
...
+        right: #{($avatar-size / 2) + 2.5rem};
```
Comment (314numberpi) [inline client/scss/components/_user-menu.scss:179]:
```suggestion
        right: #{($avatar-size / 2) + $avatar-right-margin + 0.5rem};
```

**Review by 314numberpi — CHANGES_REQUESTED** [review]
I think you would also need to change line 19 of _user-menu.scss to
```
top: $avatar-top-margin;
```
so that it doesn't get left behind.

**Comment by UncleClapton** [comment]
ah yea missed that 😆

**Review by 314numberpi — APPROVED** [review]
LGTM!
